### PR TITLE
sourcing LD_PRELOAD

### PR DESCRIPTION
**Pull request by naxingyu**

### CODE REVIEW

File: tools/extras/install_mkl.sh [hunk 1]
```diff
@@ -254,6 +254,16 @@ ConfigLdSo() {
     ldconfig )
 }
 
+(
+  set +u
```
Comment (danpovey) [inline tools/extras/install_mkl.sh]: please clarify in a comment what this does

File: tools/extras/install_mkl.sh [hunk 2]
```diff
@@ -254,6 +254,16 @@ ConfigLdSo() {
...
+
+  echo "export LD_PRELOAD:=/opt/intel/mkl/lib/intel64/libmkl_def.so:\
```
Comment (danpovey) [inline tools/extras/install_mkl.sh]: I'll merge this, but I'd like someone to remind me why we are using LD_PRELOAD and not LD_LIBRARY_PATH.

Comment (naxingyu) [reply tools/extras/install_mkl.sh]: Not sure. LD_LIBRARY_PATH just doesn't work...

Comment (danpovey) [reply tools/extras/install_mkl.sh]: so you did something like "export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/opt/intel/mkl/lib/intel64/" ?

File: tools/extras/install_mkl.sh (hunk 1 above)
Comment (naxingyu) [reply tools/extras/install_mkl.sh]: done

File: tools/extras/install_mkl.sh (hunk 2 above)
Comment (naxingyu) [reply tools/extras/install_mkl.sh]: Yes, it doesn't work.

Comment (naxingyu) [reply tools/extras/install_mkl.sh]: wait, let me try again

Comment (naxingyu) [reply tools/extras/install_mkl.sh]: yep. double checked, doesn't work.

Comment (danpovey) [reply tools/extras/install_mkl.sh]: OK.  MKL is weird.  Document the +u, why it's needed and what it does, and I'll merge.

Comment (danpovey) [reply tools/extras/install_mkl.sh]: oh, you removed it..
